common/db_test.go

```diff
@@ -578,3 +582,311 @@ func TestDBUnbondingLocks(t *testing.T) {
+	defer dbraw.Close()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	// Two tickets in the first session
+	firstSessionID := "first session"
+	_, ticket0, sig0, recipientRand0 := defaultWinningTicket(t)
+	err = dbh.StoreWinningTicket(firstSessionID, ticket0, sig0, recipientRand0)
+	if err != nil {
+		t.Fatalf("unexpected errro storing ticket: %v", err)
+	}
+
+	_, ticket1, sig1, recipientRand1 := defaultWinningTicket(t)
+	err = dbh.StoreWinningTicket(firstSessionID, ticket1, sig1, recipientRand1)
+	if err != nil {
+		t.Fatalf("unexpected errro storing ticket: %v", err)
+	}
+
+	// One ticket in the second session
+	secondSessionID := "second session"
+	_, ticket2, sig2, recipientRand2 := defaultWinningTicket(t)
+	err = dbh.StoreWinningTicket(secondSessionID, ticket2, sig2, recipientRand2)
+	if err != nil {
+		t.Fatalf("unexpected errro storing ticket: %v", err)
+	}
+
+	tickets, sigs, recipientRands, err := dbh.LoadWinningTickets(firstSessionID)
+
+	assert := assert.New(t)
+	assert.Nil(err)
+	assert.Len(tickets, 2)
+	assert.Len(sigs, 2)
+	assert.Len(recipientRands, 2)
+	assert.Equal(ticket0, tickets[0])
+	assert.Equal(sig0, sigs[0])
+	assert.Equal(recipientRand0, recipientRands[0])
+	assert.Equal(ticket1, tickets[1])
+	assert.Equal(sig1, sigs[1])
+	assert.Equal(recipientRand1, recipientRands[1])
+}
+
+func TestLoadtWinningTicket_GivenNonexistentSessionID_ReturnsEmptySlicesNoError(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	tickets, sigs, recipientRands, err := dbh.LoadWinningTickets("some sessionID")
+
+	assert := assert.New(t)
+	assert.Nil(err)
+	assert.Len(tickets, 0)
+	assert.Len(sigs, 0)
+	assert.Len(recipientRands, 0)
+}
+
+func TestLoadtWinningTicket_GivenEmptySessionID_ReturnsEmptySlicesNoError(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	tickets, sigs, recipientRands, err := dbh.LoadWinningTickets("")
+
+	assert := assert.New(t)
+	assert.Nil(err)
+	assert.Len(tickets, 0)
+	assert.Len(sigs, 0)
+	assert.Len(recipientRands, 0)
+}
+
+func defaultWinningTicket(t *testing.T) (sessionID string, ticket *pm.Ticket, sig []byte, recipientRand *big.Int) {
+	sessionID = "foo bar"
+	ticket = &pm.Ticket{
+		Sender:            randAddressOrFatal(t),
+		Recipient:         randAddressOrFatal(t),
+		FaceValue:         big.NewInt(1234),
+		WinProb:           big.NewInt(2345),
+		SenderNonce:       uint32(123),
+		RecipientRandHash: pm.RandHashOrFatal(t),
+	}
+	sig = randBytesOrFatal(42, t)
+	recipientRand = big.NewInt(4567)
+	return
+}
+
+func getRowCountOrFatal(query string, dbraw *sql.DB, t *testing.T) int {
+	var count int
+	row := dbraw.QueryRow(query)
+	err := row.Scan(&count)
+
+	if err != nil {
+		t.Fatalf("error getting db table count, cannot proceed: %v", err)
+		return -1
+	}
+
+	return count
+}
+
+func randAddressOrFatal(t *testing.T) ethcommon.Address {
```

Comment:
Should we move some of these test helpers i.e. `randBytesOrFatal`, `randBytes`, etc. into a single place to avoid duplication since `pm/helpers.go` already contains these functions?